google/yaml_validator.rb

```diff
@@ -42,17 +42,40 @@ def set_variable(value, property)
       instance_variable_set("@#{property}", value)
     end
 
-    private
+    # Does all validation checking for a particular variable.
+    # options:
+    # :default   - the default value for this variable if its nil
+    # :type      - the allowed types (single or array) that this value can be
+    # :item_type - the allowed types that all values in this array should be
```

Comment:
Can/do we error if it's not?

Reply:
Also is it required to specify an `item_type` for all arrays?

(also 👍 👍 👍 to making checking consistent!!)

Reply:
That seems like a good idea. Sounds good. I didn't make item_type required because I wanted to place as few constraints as possible on this validation.

Reply:
I'm thinking if every existing array has one, we should start by making the validation as strict as possible; if we have a really good reason to remove it later we always can, and having stricter rules will reduce the risk of us introducing bugs.

Reply:
That's fair. There's not many situations where we use Arrays with no item_type checking.

Reply:
Update: We'll have to be a little more diligent about updating the validation, but this should work just fine.

Reply:
What do you mean by this?